wandb/sdk/launch/create_job.py

```diff
@@ -0,0 +1,522 @@
+            with open(os.path.join(local_dir, "runtime.txt")) as f:
+                python_version = f.read().strip()
+        elif os.path.exists(os.path.join(local_dir, ".python-version")):
+            with open(os.path.join(local_dir, ".python-version")) as f:
+                python_version = f.read().strip().splitlines()[0]
+        else:
+            major, minor = get_current_python_version()
+            python_version = f"{major}.{minor}"
+
+    python_version = _clean_python_version(python_version)
+
+    # check if entrypoint is valid
+    if entrypoint:
+        if not os.path.exists(os.path.join(local_dir, entrypoint)):
+            wandb.termerror(f"Invalid entrypoint: {entrypoint}")
+            return None
+        # prepend working dir (specified in path)
+        entrypoint = os.path.join(src_dir, entrypoint)
+    elif not filename:
+        if path.endswith(".py"):
+            wandb.termerror("Invalid entrypoint provided in path")
+        else:
+            wandb.termerror(
+                "Invalid entrypoint. Provide an entrypoint in the path or use -E"
+            )
+        return None
+    else:  # try to make one from the path, assumes working dir is parent of file
+        rel_entrypoint = os.path.join(src_dir, filename)
+        if rel_entrypoint.startswith("/"):
+            rel_entrypoint = rel_entrypoint[1:]
+
+        if not os.path.exists(os.path.join(tempdir, rel_entrypoint)):
+            wandb.termerror(
+                f"Entrypoint {rel_entrypoint} not valid, specify one in the path or use -E"
+            )
+            return None
+        entrypoint = rel_entrypoint
+
+    # check if requirements.txt exists
+    if not os.path.exists(os.path.join(local_dir, "requirements.txt")):
+        repo_formd = path.replace(entrypoint, "")
+        wandb.termerror(
+            f"Could not find requirements.txt file in git repo at: {repo_formd}/requirements.txt"
+        )
+        return None
+
+    metadata = {
+        "git": {
+            "commit": commit,
+            "remote": ref.url,
+        },
+        "root": ref.repo,
+        "codePath": entrypoint,
+        "entrypoint": [f"python{python_version}", entrypoint],
+        "python": python_version,  # used to build container
+        "notebook": False,  # partial jobs from notebooks not supported
+    }
+
+    return metadata
+
+
+def _create_artifact_metadata(
+    path: str, entrypoint: str, runtime: Optional[str] = None
+) -> Tuple[Dict[str, Any], List[str]]:
+    if not os.path.exists(path):
+        wandb.termerror("Path must be a valid file or directory")
+        return {}, []
+
+    if not os.path.exists(os.path.join(path, "requirements.txt")):
+        wandb.termerror(f"Could not find requirements.txt file in: {path}")
+        return {}, []
+
+    # read local requirements.txt and dump to temp dir for builder
+    requirements = []
+    with open(os.path.join(path, "requirements.txt")) as f:
```

Comment:
nit: kind of weird to copy these into a list, we could just copy the requirements to the tmpdir no?

Reply:
yeah... for image jobs though we have to make a dummy requirements, which is why I made this helper. Happy to change it back though.

Reply:
Actually, maybe we rework this when we add the "accept path to requirements.txt" file feature request (from hoff) @KyleGoyette